### PR TITLE
CB-4815 Instance profiles or Roles could be filtered by API via query…

### PR DESCRIPTION
**Pull request by topolyai5**

### CODE REVIEW

File: cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsPlatformResources.java [hunk 1]
```diff
@@ -657,23 +660,14 @@ public CloudAccessConfigs accessConfigs(CloudCredential cloudCredential, Region
...
-                    properties.put("roleArn", Strings.isNullOrEmpty(roleName) ? instanceProfile.getArn() : roleName);
-                }
-                cloudAccessConfigs.getCloudAccessConfigs().add(
-                        new CloudAccessConfig(
-                                instanceProfile.getInstanceProfileName(),
-                                instanceProfile.getInstanceProfileId(),
-                                properties));
+            String accessConfigType = filters.get(ACCESS_CONFIG_TYPE);
+            Set<CloudAccessConfig> cloudAccessConfigSet;
+            if ("ROLE".equals(accessConfigType)) {
```
Comment (doktoric) [inline cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsPlatformResources.java]: Can we use an Enum for this ?

Comment (doktoric) [reply cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsPlatformResources.java]: also can we add some logs here ?

File: environment-api/src/main/java/com/sequenceiq/environment/api/v1/platformresource/PlatformResourceEndpoint.java [hunk 2]
```diff
@@ -150,7 +151,8 @@ PlatformAccessConfigsResponse getAccessConfigs(
             @QueryParam("credentialCrn") String credentialCrn,
             @QueryParam("region") String region,
             @QueryParam("platformVariant") String platformVariant,
-            @QueryParam("availabilityZone") String availabilityZone);
+            @QueryParam("availabilityZone") String availabilityZone,
+            @QueryParam("accessConfigType") AccessConfigTypeQueryParam accessConfigType);
```
Comment (doktoric) [inline environment-api/src/main/java/com/sequenceiq/environment/api/v1/platformresource/PlatformResourceEndpoint.java]: can we set default here ?

File: cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsPlatformResources.java [hunk 3]
```diff
@@ -657,23 +660,14 @@ public CloudAccessConfigs accessConfigs(CloudCredential cloudCredential, Region
...
+                cloudAccessConfigSet = getAccessConfigByRole(client);
+            } else {
+                cloudAccessConfigSet = getAccessConfigByInstanceProfile(client);
             }
+            cloudAccessConfigs.getCloudAccessConfigs().addAll(cloudAccessConfigSet);
         } catch (AmazonServiceException ase) {
             if (ase.getStatusCode() == UNAUTHORIZED) {
                 String policyMessage = "Could not get instance profile roles because the user does not have enough permission.";
```
Comment (doktoric) [inline cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsPlatformResources.java]: I think the message here is not correct

File: cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsPlatformResources.java (hunk 1 above)
Comment (keyki) [reply cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsPlatformResources.java]: We should also add a log if the type is not ROLE or null then we will return all instance profiles

**Pull request merged**